### PR TITLE
deno test should support directories as arguments

### PR DESCRIPTION
**Pull request by ry**
Fixes https://github.com/denoland/deno/issues/3999

### CODE REVIEW

File: cli/fs.rs [hunk 1]
```diff
@@ -188,3 +188,26 @@ mod tests {
     assert_eq!(resolve_from_cwd(expected).unwrap(), expected);
   }
 }
+
+pub fn files_in_subtree<F>(root: PathBuf, filter: F) -> Vec<PathBuf>
+where
+  F: Fn(&Path) -> bool,
+{
+  // TODO(ry) Use WalkDir instead of globs.
```
Comment (bartlomieju) [inline cli/fs.rs]: Maybe even [`fs::read_dir`](https://doc.rust-lang.org/std/fs/fn.read_dir.html)?

Comment (ry) [reply cli/fs.rs]: It's not recursive, but yes some recursive version of that.

File: cli/test_runner.rs [hunk 2]
```diff
@@ -3,12 +3,21 @@
...
+use std::path::Path;
 use std::path::PathBuf;
 use url::Url;
 
+fn test_file_filter(p: &Path) -> bool {
+  let s = p.as_os_str().to_string_lossy();
+  s.ends_with("_test.ts")
+    || s.ends_with("_test.js")
+    || s.ends_with(".test.ts")
+    || s.ends_with(".test.js")
```
Comment (nayeemrmn) [inline cli/test_runner.rs]:
```rust
    || s == "test.ts"
    || s == "test.js"
```

File: cli/test_runner.rs [hunk 3]
```diff
@@ -3,12 +3,18 @@
 use crate::installer::is_remote_url;
 use deno_core::ErrBox;
 use std;
+use std::path::Path;
 use std::path::PathBuf;
 use url::Url;
 
+fn is_supported(p: &Path) -> bool {
+  let s = p.as_os_str().to_string_lossy();
+  s.ends_with("test.ts") || s.ends_with("test.js")
```
Comment (nayeemrmn) [inline cli/test_runner.rs]: This is too loose.

Comment (ry) [reply cli/test_runner.rs]: tightened

**Comment by ry** [comment]
closed in favor of https://github.com/denoland/deno/pull/4011

**Pull request closed without merging**